core/server/src/main/java/alluxio/worker/AlluxioWorker.java

```diff
@@ -39,6 +41,12 @@ public static void main(String[] args) {
       System.exit(-1);
     }
 
+    if (!Configuration.containsKey(PropertyKey.MASTER_HOSTNAME)) {
```

Comment:
The master hostname isn't required by the master process. If unset it will default to the local host name.